core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -391,6 +412,46 @@ void setupVersionMetrics() {
              .register(meterRegistry);
     }
 
+    /**
+     * Sets up gauge metric for each server certificate.
+     */
+    private void setupTlsMetrics(SslContext sslContext, String hostname) {
+        final MeterRegistry meterRegistry = config().meterRegistry();
+
+        final SSLSession sslSession = validateSslContext(sslContext);
+        for (Certificate certificate : sslSession.getLocalCertificates()) {
+            try {
+                final X509Certificate x509Certificate = (X509Certificate) certificate;
+                final String commonName = getCommonName(x509Certificate);
+
+                Gauge.builder("armeria.server.certificate.validity", x509Certificate, x509Cert -> {
+                         try {
+                             x509Cert.checkValidity();
+                         } catch (CertificateExpiredException | CertificateNotYetValidException e) {
+                             return 0;
+                         }
+                         return 1;
+                     })
+                     .description("1 if certificate is in validity period, 0 if certificate is not in " +
+                                  "validity period")
+                     .tags("common_name", commonName, "hostname", hostname)
```

Comment:
```suggestion
                     .tags("common.name", commonName, "hostname", hostname)
```
(Micrometer convention)